docs/compilers/CSharp/Compiler Breaking Changes - post VS2017.md

```diff
@@ -77,3 +77,19 @@ In Visual Studio 2017 version 15.6 such conversions will be explicitly disallowe
 Example: `Func<int> f = default(TypedReference).GetHashCode; // new error CS0123: No overload for 'GetHashCode' matches delegate 'Func<int>'` 
    
 - https://github.com/dotnet/roslyn/pull/23416 Before Visual Studio 2017 version 15.6 (Roslyn version 2.8) the compiler accepted `__arglist(...)` expressions with void-typed arguments. For instance, `__arglist(Console.WriteLine())`. But such program would fail at runtime. In Visual Studio 2017 version 15.6, this causes a compile-time error.
+
+- Visual Studio 2017 15.0-15.5 shipped with a bug around definite assignment of local functions that did not produce definite assignment errors when an uncalled local function contains a nested lambda which captures a variable. For example:
+    ```csharp
+    void Method()
+    {
+        void Local()
+        {
+            Action a = () =>
+            {
+                int x;
+                x++; // No error in 15.0 - 15.5
+            };
+        }
+    }
+    ```
+    This is changed to now produce an error that the variable is not definitely assigned.
```

Comment:
In ... 15.6, this changed to produce...

Reply:
I think we should make the specific version change later. There's no guarantee this actually ships in 15.6.

Reply:
This PR is for master branch, which is 15.6, and you are about to merge...